packages/nimble-components/src/rich-text-mention/users/tests/rich-text-mention-users.spec.ts

```diff
@@ -0,0 +1,348 @@
+            await connect();
+            expect(element.checkValidity()).toBeTrue();
+            expect(element.validity.unsupportedMappingType).toBeFalse();
+            expect(element.validity.duplicateMappingMentionHref).toBeFalse();
+            expect(element.validity.missingMentionHrefValue).toBeFalse();
+            expect(element.validity.unsupportedMentionHrefValue).toBeFalse();
+        });
+
+        it('is valid with valid unique string key values', async () => {
+            ({ element, connect, disconnect } = await setup(
+                [
+                    { key: 'user:1', displayName: 'user' },
+                    { key: 'user:2', displayName: 'user' },
+                    { key: 'user:3', displayName: 'user' },
+                    { key: 'user:4', displayName: 'user' },
+                    { key: 'user:5', displayName: 'user' }
+                ],
+                ''
+            ));
+            await connect();
+            expect(element.checkValidity()).toBeTrue();
+            expect(element.validity.unsupportedMappingType).toBeFalse();
+            expect(element.validity.duplicateMappingMentionHref).toBeFalse();
+            expect(element.validity.missingMentionHrefValue).toBeFalse();
+            expect(element.validity.unsupportedMentionHrefValue).toBeFalse();
+            expect(element.validity.missingDisplayNameValue).toBeFalse();
+        });
+
+        it('is invalid with missing key', async () => {
+            ({ element, connect, disconnect } = await setup([
+                { displayName: 'user' },
+                { key: 'user:1', displayName: 'user' }
+            ]));
+            await connect();
+            expect(element.checkValidity()).toBeFalse();
+            expect(element.validity.missingMentionHrefValue).toBeTrue();
+            expect(element.mentionInternals.mentionConfig).toBe(undefined);
+        });
+
+        it('is invalid with missing displayName', async () => {
+            ({ element, connect, disconnect } = await setup([
+                { key: 'user:2' },
+                { key: 'user:1', displayName: 'user' }
+            ]));
+            await connect();
+            expect(element.checkValidity()).toBeFalse();
+            expect(element.validity.missingDisplayNameValue).toBeTrue();
+            expect(element.mentionInternals.mentionConfig).toBe(undefined);
+        });
+
+        it('is invalid with duplicate key values', async () => {
+            ({ element, connect, disconnect } = await setup([
+                { key: 'user:1', displayName: 'user' },
+                { key: 'user:1', displayName: 'user' }
+            ]));
+            await connect();
+            expect(element.checkValidity()).toBeFalse();
+            expect(element.validity.duplicateMappingMentionHref).toBeTrue();
+            expect(element.mentionInternals.mentionConfig).toBe(undefined);
+        });
+
+        it('is invalid with mismatch pattern', async () => {
```

Comment:
This doesn't seem like a different case than the next test ('is invalid with invalid mismatching key'). This one's pattern says "invalidPattern", but it's not really invalid, is it? It's just that the keys don't match it as a regex? Unless I'm missing something, I think you can remove this test.

Reply:
Removed this case.